### PR TITLE
Use sprintf around formatted strings

### PR DESCRIPTION
**Pull request by Mezzle**

### CODE REVIEW

**Comment by samsonasik** [comment]
:+1:

File: src/SaveHandler/MongoDB.php [hunk 1]
```diff
@@ -57,8 +57,10 @@ public function __construct($mongo, MongoDBOptions $options)
     {
         if (!($mongo instanceof \MongoClient || $mongo instanceof \Mongo)) {
             throw new InvalidArgumentException(
-                'Parameter of type %s is invalid; must be MongoClient or Mongo',
-                (is_object($mongo) ? get_class($mongo) : gettype($mongo))
+                sprintf(
+                    'Parameter of type %s is invalid; must be MongoClient or Mongo',
+                    (is_object($mongo) ? get_class($mongo) : gettype($mongo))
+                )
```
Comment (Martin-P) [inline src/SaveHandler/MongoDB.php:65]: ZF2 uses a slightly different indentation:

``` php
throw new InvalidArgumentException(sprintf(
    'Parameter of type %s is invalid; must be MongoClient or Mongo',
    (is_object($mongo) ? get_class($mongo) : gettype($mongo))
));
```

Comment (Mezzle) [inline src/SaveHandler/MongoDB.php:65]: Where is this documented?

Comment (Martin-P) [inline src/SaveHandler/MongoDB.php:65]: > Where is this documented?

Good question for a framework where IMHO documentation is a problem :smile:

Looking at other ZF2 code in almost every situation the indentation I mentioned is used: https://github.com/zendframework/zend-session/search?utf8=%E2%9C%93&q=Exception%28sprintf%28

Comment (samsonasik) [inline src/SaveHandler/MongoDB.php:65]: I think it may for consistency instead of standard

Comment (Mezzle) [inline src/SaveHandler/MongoDB.php:65]: Personally, I think the way you've suggested indenting it is what _caused_ this error to occur.  It's obvious to me that someone's written their code, and copy paste/modified the arguments within, and not noticed the `sprintf` tucked away :)

Comment (Martin-P) [inline src/SaveHandler/MongoDB.php:65]: The error is caused by the fact the original PR was merged without proper tests: https://github.com/zendframework/zf2/pull/3221 If there was a unittest for this exception the problem would have been detected immediately, that is why it is important to add unittests. BTW a unittest is missing for this PR too :wink:

**Comment by mwillbanks** [comment]
Nice catch :) Yes; we certainly need some tests around this guy.

**Comment by Mezzle** [comment]
Added Tests, Coveralls doesn't get it because `setUp` skips tests if no Mongo extension.

File: test/SaveHandler/MongoDBTest.php [hunk 2]
```diff
@@ -69,6 +76,20 @@ public function tearDown()
         }
     }
 
+    public function testConstructorThrowsException()
+    {
+        $notMongo = new \stdClass();
+
+        $expectedExceptionMessage =  sprintf(
+            'Parameter of type %s is invalid; must be MongoClient or Mongo',
+            get_class($notMongo)
```
Comment (Maks3w) [inline test/SaveHandler/MongoDBTest.php]: Expectation should be static and not be built with the same logic as the implementation

**Comment by Maks3w** [comment]
:+1: For add the test

File: test/SaveHandler/MongoDBTest.php [hunk 3]
```diff
@@ -43,18 +43,25 @@ class MongoDBTest extends \PHPUnit_Framework_TestCase
...
-        $this->options = new MongoDBOptions([
-            'database' => 'zf2_tests',
-            'collection' => 'sessions',
-        ]);
+        $this->options = new MongoDBOptions(
+            [
+                'database' => 'zf2_tests',
+                'collection' => 'sessions',
+            ]
+        );
```
Comment (Martin-P) [inline test/SaveHandler/MongoDBTest.php]: Previous indentation was the right one.

Comment (Maks3w) [inline test/SaveHandler/MongoDBTest.php]: Indeed however I'll preffer don't add more noise in  the PR for just cosmetic changes.

Comment (Martin-P) [inline test/SaveHandler/MongoDBTest.php]: No problem. I guess it will be corrected when it is merged?

Comment (Maks3w) [inline test/SaveHandler/MongoDBTest.php]: no. The only wrong thing is to change unrelated lines in the PR. Both indentations are correct.

Comment (Martin-P) [inline test/SaveHandler/MongoDBTest.php]: Oops, sorry, I thought your comment was to me, but it was for @Mezzle

Comment (Maks3w) [inline test/SaveHandler/MongoDBTest.php]: (both)
